src/test/java/TrainTest.java

```diff
@@ -0,0 +1,36 @@
+
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
+public class TrainTest {
+
+    Train t1, t2, t3;
+
+    @Before
+    public void setUp() throws Exception {
+        t1 = new Train();
+        t2 = new Train(100);
+        t3 = new Train(500, 12345678);
+    }
+
+    @Test(timeout = 50)
+    public void TestGetMaxSpeed() {
+        assertEquals(50, t1.getMaxSpeed());
+        assertEquals(100, t2.getMaxSpeed());
+        assertEquals(500, t3.getMaxSpeed());
+    }
+
+    @Test(timeout = 50)
+    public void TestUpgradeSpeed() {
+        t1.upgradeSpeed();
```

Comment:
Probably should also test at least t2 as this one takes maxSpeed as an input unlike t1.